src/plone/restapi/services/querystringsearch/get.py

```diff
@@ -38,6 +48,13 @@ def reply(self):
             limit=limit,
         )
 
+        if SUPPORT_NOT_UUID_QUERIES and not (IPloneSiteRoot.providedBy(self.context)):
```

Comment:
Maybe a code comment: "# Exclude "self" content item from the results when ZCatalog supports NOT UUID queries and is called on a content object." Now that I read it, maybe we should do the IPloneSiteRoot check first and amend the comment accordingly. Though that's nitpicking, we can do that after the merge as well.

Reply:
Done!